webapp/app/controllers/test.js

```diff
@@ -0,0 +1,69 @@
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  treeCompliantMetadata: function() {
+    var model = this.get('model');
+
+    var metadata = model.get('testMetadata');
+    var compliantObject = [];
+    this.createSubtree(compliantObject, '', metadata);
+    return compliantObject;
+  }.property('model.testMetadata'),
+
+  createSubtree: function myself(array_to_append, parentIndex, treeObject) {
```

Comment:
so what you are saying is that the entire 
```
<table class="table table-bordered tree">
        <tbody>
        {{#each metadata in treeCompliantMetadata}}
...
        {{/each}}
        </tbody>
    </table>
```
Should be inside a component? or something with smaller granularity?